src/Components/People.js

```diff
@@ -0,0 +1,62 @@
+import axios from "axios";
+import { Component } from "react";
+
+class People extends Component {
+    state = {people:[], input:"", notFound: false, person: {}}
+
+    fetchPeople = async () => {
+        try {
+            const res = await axios.get("https://ghibliapi.herokuapp.com/people/");
+            this.setState({ people: res.data.map((persons) => {
+                persons = {
+                    name: persons.name,
+                    age: persons.age,
+                    gender: persons.gender
+                }
+            }) });
+          //   debugger
+          } catch (error) {
+            this.setState({ people: [] });
+          }
+    }
+
+    componentDidMount() {
+        this.fetchPeople()
+    }
+
+    handleSubmit = (e) => {
+        e.preventDefault();
+        const {input, people} = this.state
+        people.forEach((person) => {
+            if (person.name === input) {
+                this.setState({person})
+            } else {
+                this.setState({notFound:true})
+            }
+        })
```

Comment:
forEach will iterate through each person. This might work for you but not for the reasons you want. At the moment, let's the first iteration was the person, we set the state for person, but then the second iteration happens and notFound is set to true.